### PR TITLE
Sync with ballerina-0.991.0

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
Update the connector with ballerina-0.991.0

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: gsheets4/gsheets_endpoint.bal [hunk 1]
```diff
@@ -461,10 +461,6 @@ public remote function Client.setSheetValues(string spreadsheetId, string sheetN
 
 public function Client.init(SpreadsheetConfiguration spreadsheetConfig) {
```
Comment (ldclakmal) [inline gsheets4/gsheets_endpoint.bal]: We don't need this function with the new way of configuration. Right?

File: gsheets4/tests/test.bal [hunk 2]
```diff
@@ -24,10 +24,18 @@ SpreadsheetConfiguration spreadsheetConfig = {
...
-            clientSecret: config:getAsString("CLIENT_SECRET"),
-            refreshToken: config:getAsString("REFRESH_TOKEN")
+            config: {
+                grantType: http:DIRECT_TOKEN,
+                config: {
+                    accessToken: config:getAsString("ACCESS_TOKEN"),
+                    refreshConfig: {
+                        clientId: config:getAsString("CLIENT_ID"),
+                        clientSecret: config:getAsString("CLIENT_SECRET"),
+                        refreshUrl: "https://www.googleapis.com/oauth2/v3/token",
```
Comment (ldclakmal) [inline gsheets4/tests/test.bal]: Shall we move this to config?

File: gsheets4/gsheets_types.bal [hunk 3]
```diff
@@ -132,7 +132,7 @@ public function Spreadsheet.getSheetByName(string sheetName) returns Sheet|error
         return err;
     } else {
         foreach var sheet in sheets {
-            if (sheet.properties != null) {
+            if (sheet.properties != ()) {
```
Comment (MaryamZi) [inline gsheets4/gsheets_types.bal]: `sheet.properties` can never be `()` right? Shall we check and fix this?

File: gsheets4/gsheets_types.bal [hunk 4]
```diff
@@ -70,34 +70,34 @@ public type Sheet record {
...
     int columnCount = 0;
     int frozenRowCount = 0;
     int frozenColumnCount = 0;
     boolean hideGridlines = false;
-};
+|};
 
 //Functions binded to Spreadsheet struct
 public function Spreadsheet.getSpreadsheetName() returns string|error {
     string title = "";
```
Comment (MaryamZi) [inline gsheets4/gsheets_types.bal]: Do we need this variable?

File: gsheets4/gsheets_types.bal [hunk 5]
```diff
@@ -70,34 +70,34 @@ public type Sheet record {
...
-    if (self.properties == null) {
+    if (self.properties.length() == 0) {
```
Comment (MaryamZi) [inline gsheets4/gsheets_types.bal]: `self.properties` is a record with fields with default values. So length will never be 0 right?

I think we can replace L99-L105 with `return self.properties.title;` unless we need to check for an empty string for title like https://github.com/wso2-ballerina/module-googlespreadsheet/pull/50/files#diff-8aa445f43832f67ddbe3d9dab67269bdR110.

File: gsheets4/gsheets_types.bal [hunk 6]
```diff
@@ -132,7 +132,7 @@ public function Spreadsheet.getSheetByName(string sheetName) returns Sheet|error
         return err;
     } else {
         foreach var sheet in sheets {
-            if (sheet.properties != null) {
+            if (sheet.properties.length() != 0) {
```
Comment (MaryamZi) [inline gsheets4/gsheets_types.bal]: Again, not needed right?

File: gsheets4/gsheets_types.bal [hunk 7]
```diff
@@ -70,35 +70,34 @@ public type Sheet record {
...
-};
+|};
 
 //Functions binded to Spreadsheet struct
 public function Spreadsheet.getSpreadsheetName() returns string|error {
-    string title = "";
-    if (self.properties == null) {
-        error err = error(SPREADSHEET_ERROR_CODE, { message: "Unable to find the spreadsheet properties" });
+    if (self.properties.title == EMPTY_STRING) {
+        error err = error(SPREADSHEET_ERROR_CODE, { message: "Unable to find the spreadsheet title" });
```
Comment (MaryamZi) [inline gsheets4/gsheets_types.bal]: Please create an issue to improve these error messages. We may have to validate if the response could ever have an empty string as the title.

File: gsheets4/gsheets_types.bal [hunk 8]
```diff
@@ -132,7 +131,7 @@ public function Spreadsheet.getSheetByName(string sheetName) returns Sheet|error
         return err;
     } else {
         foreach var sheet in sheets {
-            if (sheet.properties != null) {
+            if (sheet.properties.title == EMPTY_STRING) {
```
Comment (MaryamZi) [inline gsheets4/gsheets_types.bal]: Shouldn't we be checking the opposite? :) Don't we have tests for these functions?

I think we can remove this check all together and only have the next if condition.

File: gsheets4/gsheets_types.bal (hunk 7 above)
Comment (kalaiyarasiganeshalingam) [reply gsheets4/gsheets_types.bal]: Issue https://github.com/wso2-ballerina/module-googlespreadsheet/issues/51

**Pull request merged**